From: Simon McVittie <smcv@collabora.com>
Date: Thu, 31 Aug 2017 12:21:18 +0100
Subject: tests: Don't require user xattrs

Since Flatpak 0.9.6, we use libostree in bare-user-only mode, which
does not rely on xattrs.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 0.9.9, commit:a53a752a354a569b8cd4a1bb19e2cbfd837aac12
---
 tests/libtest.sh                          | 8 --------
 tests/test-bundle.sh                      | 1 -
 tests/test-extensions.sh                  | 1 -
 tests/test-oci.sh                         | 1 -
 tests/test-repo.sh                        | 1 -
 tests/test-run.sh                         | 1 -
 tests/test-unsigned-summaries.sh          | 1 -
 tests/test-update-remote-configuration.sh | 1 -
 8 files changed, 15 deletions(-)

diff --git a/tests/libtest.sh b/tests/libtest.sh
index 77da52c..96f8b2a 100644
--- a/tests/libtest.sh
+++ b/tests/libtest.sh
@@ -309,14 +309,6 @@ run_sh () {
     ${CMD_PREFIX} flatpak run --command=bash ${ARGS-} org.test.Hello -c "$*"
 }
 
-skip_without_user_xattrs () {
-    touch ${TEST_DATA_DIR}/test-xattrs
-    if ! setfattr -n user.testvalue -v somevalue ${TEST_DATA_DIR}/test-xattrs; then
-        echo "1..0 # SKIP this test requires xattr support"
-        exit 0
-    fi
-}
-
 skip_without_bwrap () {
     if [ -z "${FLATPAK_BWRAP:-}" ]; then
         # running installed-tests: assume we know what we're doing
diff --git a/tests/test-bundle.sh b/tests/test-bundle.sh
index 6eec641..7fba235 100755
--- a/tests/test-bundle.sh
+++ b/tests/test-bundle.sh
@@ -22,7 +22,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 
 echo "1..7"
 
diff --git a/tests/test-extensions.sh b/tests/test-extensions.sh
index 3597b16..2a13a86 100755
--- a/tests/test-extensions.sh
+++ b/tests/test-extensions.sh
@@ -22,7 +22,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 
 echo "1..2"
 
diff --git a/tests/test-oci.sh b/tests/test-oci.sh
index 82befed..06effca 100755
--- a/tests/test-oci.sh
+++ b/tests/test-oci.sh
@@ -22,7 +22,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 
 echo "1..6"
 
diff --git a/tests/test-repo.sh b/tests/test-repo.sh
index 763405e..a5b54d2 100755
--- a/tests/test-repo.sh
+++ b/tests/test-repo.sh
@@ -22,7 +22,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 
 if [ x${USE_COLLECTIONS_IN_CLIENT-} == xyes ] || [ x${USE_COLLECTIONS_IN_SERVER-} == xyes ] ; then
     skip_without_p2p
diff --git a/tests/test-run.sh b/tests/test-run.sh
index b9e2e7a..c19dbe1 100755
--- a/tests/test-run.sh
+++ b/tests/test-run.sh
@@ -22,7 +22,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 
 echo "1..12"
 
diff --git a/tests/test-unsigned-summaries.sh b/tests/test-unsigned-summaries.sh
index 2fccc71..ff14af2 100755
--- a/tests/test-unsigned-summaries.sh
+++ b/tests/test-unsigned-summaries.sh
@@ -25,7 +25,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 skip_without_p2p
 
 echo "1..7"
diff --git a/tests/test-update-remote-configuration.sh b/tests/test-update-remote-configuration.sh
index 76957ba..df0a8ea 100755
--- a/tests/test-update-remote-configuration.sh
+++ b/tests/test-update-remote-configuration.sh
@@ -25,7 +25,6 @@ set -euo pipefail
 . $(dirname $0)/libtest.sh
 
 skip_without_bwrap
-skip_without_user_xattrs
 skip_without_p2p
 
 echo "1..3"
